db/Pos/DB/Class.hs

```diff
@@ -104,13 +106,29 @@ class (HasConfiguration, MonadBaseControl IO m, MonadThrow m) => MonadDBRead m w
         , Bi (IterValue i)
         ) => DBTag -> Proxy i -> Source (ResourceT m) (IterType i)
 
+    -- | Get block by header hash
+    dbGetRawBlock :: HeaderHash -> m (Maybe RawBlock)
+
+    -- | Get undo by header hash
+    dbGetRawUndo :: HeaderHash -> m (Maybe RawUndo)
+
 instance {-# OVERLAPPABLE #-}
     (MonadDBRead m, MonadTrans t, MonadThrow (t m), MonadBaseControl IO (t m)) =>
         MonadDBRead (t m)
   where
     dbGet tag = lift . dbGet tag
     dbIterSource tag (p :: Proxy i) =
         hoist (hoist lift) (dbIterSource tag p)
+    dbGetRawBlock = lift . dbGetRawBlock
+    dbGetRawUndo = lift . dbGetRawUndo
+
+
+type MonadBlockDBRead m = (MonadDBRead m, BlockchainHelpers MainBlockchain)
```

Comment:
Can you please point out where do we use `BlockchainHelpers` instance? I don't really see it.

Reply:
when deserialize `Block` to validate validness